### DISCUSSION TITLE
VTOL autoland

### DISCUSSION THREAD

**Discussion by ivolanski** [iNavFlight/inav#10400] [General]
Hello, I am flying some VTOL with INAV and I really enjoy to have RTH as it is, I use it a lot more than loiter because it can go directly to the altitude I want, so I can put my goggles with no worries. But I would like to be able to set a stick and have the aircraft going from airplane to tricopter and landing on home position. Is this or something similar possible?

The main reason that I would like this is that I fly at night and most of the time it is impossible to see the environment and land manually. At this moment I am using a arduplane VTOL for this night fights but I really dislike that user unfriendly interface and would be much better to just stay with INAV if I could manage to make something like this to work.

Thanks

**Comment by Jetrell** [comment]
Have you seen this in the docs https://github.com/iNavFlight/inav/blob/master/docs/VTOL.md#automated-switching-rth-optional

**Reply by ivolanski** [reply]
> Have you seen this in the docs https://github.com/iNavFlight/inav/blob/master/docs/VTOL.md#automated-switching-rth-optional

Hi, yes I saw it, thanks. But my point is that I really enjoy RTH without land because it's a great way to put the bird loitering above me, always at a fixed altitude. What I am searching for is a way to keep the RTH loitering behavior and add a second function for auto land. 
Another option could be to change the RTH to auto land and create a new loitering function that has a configured altitude. Maybe programing tab? Maybe some function that I am missing?

Thanks anyway, that information is great.

**Reply by sensei-hacker** [reply]
> Hi, yes I saw it, thanks. But my point is that I really enjoy RTH without land because it's a great way to put the bird loitering above me, always at a fixed altitude. What I am searching for is a way to keep the RTH loitering behavior and add a second function for auto land. Another option could be to change the RTH to auto land and create a new loitering function that has a configured altitude. Maybe programing tab? Maybe some function that I am missing?

Without delving into the programming tab, you can RTH to loiter (at the set loiter altitude), then have a switch that transitions to MC in position mode.  Lower throttle just a bit and it will go straight down.

**Reply by ivolanski** [reply]
> > Hi, yes I saw it, thanks. But my point is that I really enjoy RTH without land because it's a great way to put the bird loitering above me, always at a fixed altitude. What I am searching for is a way to keep the RTH loitering behavior and add a second function for auto land. Another option could be to change the RTH to auto land and create a new loitering function that has a configured altitude. Maybe programing tab? Maybe some function that I am missing?
> 
> Without delving into the programming tab, you can RTH to loiter (at the set loiter altitude), then have a switch that transitions to MC in position mode. Lower throttle just a bit and it will go straight down.

This is something that I tried before but it will not land at home point, it will land where transition ended that should be a loiter distance radius from home, or actually, even more far, because the forward momentum during transition.

**Reply by Jetrell** [reply]
> But my point is that I really enjoy RTH without land because it's a great way to put the bird loitering above me, always at a fixed altitude. What I am searching for is a way to keep the RTH loitering behavior and add a second function for auto land.

Or you could set `nav_rth_allow_landing = FS` in place of `ALWAYS`..  This will keep it hovering above the home location. And you could place the Failsafe mode on a switch, to trigger a landing when required.

**Reply by ivolanski** [reply]
> > But my point is that I really enjoy RTH without land because it's a great way to put the bird loitering above me, always at a fixed altitude. What I am searching for is a way to keep the RTH loitering behavior and add a second function for auto land.
> 
> Or you could set `nav_rth_allow_landing = FS` in place of `ALWAYS`.. This will keep it hovering above the home location. And you could place the Failsafe mode on a switch, to trigger a landing when required.

Yes, right. This is what I am searching for, there is just one problem. Sometimes during auto land we need to take over because things happens. Today I was doing auto land and a wind gust somehow managed to make the aircraft start to bounce left and right even after the wind gust, I have no idea why. But anyway, I just took over and did a manual landing. So FS sounds like too dangerous. I would like to have a switch to change some configuration to perform the auto land. Maybe something that could change set mixer_automated_switch= ON
